src/App.tsx

```diff
@@ -15,48 +17,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SORT_BY_ALPHABET = 'Sort alphabetically',
+  SORT_BY_LENGTH = 'Sort by length',
+}
+
+interface SortParams {
+  sortField: SortType;
+  reverse: boolean;
+}
+
+function getPrepareGoods(goods: string[], { sortField, reverse }: SortParams) {
+  let prepareGoods = [...goods];
+
+  if (sortField) {
+    prepareGoods = prepareGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.SORT_BY_ALPHABET:
+          return good1.localeCompare(good2);
+        case SortType.SORT_BY_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    return prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortParams, setSortParams] = useState<SortParams>({
+    sortField: SortType.SORT_BY_ALPHABET,
+    reverse: false,
```

Comment:
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig
```suggestion
    isReversed: false,
```